JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java

```diff
@@ -146,31 +136,160 @@ private int getUserId(String userNickName) {
-                .setParameter("nickname", userNickName)
-                .getSingleResult();
+        User user;
+        try {
+            user = (User) em.createQuery("SELECT p FROM User p WHERE p.nickName=:nickname")
+                    .setParameter("nickname", userNickName)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETACCESS()", noResult);
+            return null;
+        }
 
         return user;
     }
 
-    private Access getAccess(String accessRights) {
+    public Access getAccess(String accessRights) {
+        Access access;
 
-        Access access = (Access)em.createQuery("SELECT p.accessId FROM Access p WHERE p.accessName =:accessrights")
-                .setParameter("accessrights", accessRights)
-                .getSingleResult();
+        try {
+            access = (Access) em.createQuery("SELECT p FROM Access p WHERE p.accessName =:accessname")
+                    .setParameter("accessname", accessRights)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETACCESS()", noResult);
+            return null;
+        }
         return access;
     }
 
-    private Group getGroup(int groupId) {
+    public Group getGroup(int groupId) {
+
+        Group group = null;
+        try {
+            group = (Group) em.createQuery("SELECT g FROM Group g WHERE g.groupId=:groupid")
+                    .setParameter("groupid", groupId)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETGROUP()", noResult);
+            return null;
+        }
+
+        return group;
+    }
+
+    public Group getGroup(String groupName) {
+
+        Group group = null;
+        try {
+            group = (Group) em.createQuery("SELECT g FROM Group g WHERE g.groupName=:groupname")
+                    .setParameter("groupname", groupName)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETGROUP()", noResult);
+            return null;
+        }
+        return group;
+    }
+
+    public int getGroupId(String groupName) {
+
+        Group group = getGroup(groupName);
+        if (group == null) {
+            return -1;
+        }
+        return group.getGroupId();
+
+    }
+
+    public Group createGroup(String groupName) {
+
+        Group g = new Group(groupName);
+
+        em.persist(g);
+        em.flush();
+
+        return g;
+
+
+    }
+
+    public Access createAccess(String accessName) {
+        Access a = new Access(accessName);
+
+        em.persist(a);
+        em.flush();
+
+        return a;
+
+    }
+
+    public UserGroup createUserGroup(User u, Group g, Access a) {
```

Comment:
It was a method for testing , can be deleted
23 марта 2016 г., в 2:24, alexander1leonov <notifications@github.com> написал(а):

> In JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java:
> 
> > +        return g;
> > +
> > +
> > +    }
> > +
> > +    public Access createAccess(String accessName) {
> > +        Access a = new Access(accessName);
> > +
> > +        em.persist(a);
> > +        em.flush();
> > +
> > +        return a;
> > +
> > +    }
> > +
> > +    public UserGroup createUserGroup(User u, Group g, Access a) {
> Why do we need both addMember() and this method?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>